### ISSUE TITLE
Execution is launched in CI, despite error in code generation

### ISSUE THREAD

**Issue by ChadliaJerad** [lf-lang/lingua-franca#1949]
There is a code generation error [here](https://github.com/lf-lang/lingua-franca/actions/runs/5871228264/job/15920103516#step:7:860), as I forgot the `self` construct to refer to a state variable ([here](https://github.com/lf-lang/lingua-franca/actions/runs/5871228264/job/15920103516#step:7:853)).

The [execution is launched](https://github.com/lf-lang/lingua-franca/actions/runs/5871228264/job/15920103516#step:7:887) though, causing the RTI to just wait until test timeout.

**Comment by lhstrh** [comment]
I agree, this is a bug in the test framework. We should not be launching the test if code generation or compilation failed.
